cmd/zrepl/zrepl.c

```diff
@@ -877,6 +965,18 @@ uzfs_zvol_io_ack_sender(void *arg)
 		ZREPL_LOG("ACK for op:%d with seq-id %ld\n",
 		    zio_cmd->hdr.opcode, zio_cmd->hdr.io_seq);
 
+		/* account for space taken by metadata headers */
+		if (zio_cmd->hdr.opcode == ZVOL_OPCODE_READ) {
+			md_len = 0;
+			for (metadata_desc_t *md = zio_cmd->metadata_desc;
```

Comment:
need to handle case of metadata_desc being NULL. md_len should be set to 1 in such case

Reply:
will fix 👍